doc/user/methods/gradient_boosting.md

```diff
@@ -0,0 +1,241 @@
+
+### Other Functionality
+
+ * `WeakLearner(size_t i)` returns a [`DecisionTreeRegressor&`]
+    (decision_tree_regressor.md) indicating which weak learner 
+    is being used. 
+
+ * `gb.NumWeakLearners()` returns a `size_t` indicating how 
+    many weak learners are being used by the model.
+    - `gb.NumWeakLearners(x)` will set the number of weak learners 
+    that will be used the *next time* `Train()` is called to `x`.
+
+ * `gb.NumClasses()` returns a `size_t` indicating the number of classes the
+   model was trained on.
+
+For complete functionality, the [source
+code](/src/mlpack/methods/grad_boosting/grad_boosting.hpp) can be consulted.
+Each method is fully documented.
+
+### Simple Examples
+
+See also the [simple usage example](#simple-usage-example) for a trivial use of
+`GradBoosting`.
+
+---
+
+Train a gradient boosting model on mixed categorical data, passing 
+arguments into the Train function:
+
+```c++
+// Load a categorical dataset.
+arma::mat dataset;
+mlpack::data::DatasetInfo info;
+// See https://datasets.mlpack.org/covertype.train.arff.
+mlpack::data::Load("covertype.train.arff", dataset, info, true);
+
+arma::Row<size_t> labels;
+// See https://datasets.mlpack.org/covertype.train.labels.csv.
+mlpack::data::Load("covertype.train.labels.csv", labels, true);
+
+// Create the model object.
+mlpack::GradBoosting gb;
+// Train on the given dataset, specifying the number of weak learners as 5.
+gb.Train(dataset, info, labels, 7 /* classes */, 5 /* number of weak learners */);
+
+// Load categorical test data.
+arma::mat testDataset;
+// See https://datasets.mlpack.org/covertype.test.arff.
+mlpack::data::Load("covertype.test.arff", testDataset, true);
+
+// Predict class of first test point.
+const size_t firstPrediction = gb.Classify(testDataset.col(0));
+std::cout << "Predicted class of first test point is " << firstPrediction << "."
+    << std::endl;
+
+```
+
+---
+
+Train a gradient boosting model on mixed categorical data, passing 
+arguments into the Constructor function:
+
+```c++
+// Load a categorical dataset.
+arma::mat dataset;
+mlpack::data::DatasetInfo info;
+// See https://datasets.mlpack.org/covertype.train.arff.
+mlpack::data::Load("covertype.train.arff", dataset, info, true);
+
+arma::Row<size_t> labels;
+// See https://datasets.mlpack.org/covertype.train.labels.csv.
+mlpack::data::Load("covertype.train.labels.csv", labels, true);
+
+// Create the model object.
+mlpack::GradBoosting gb(dataset, info, labels, 
+   7 /* classes */, 5 /* number of weak learners */);
+
+// Load categorical test data.
+arma::mat testDataset;
+// See https://datasets.mlpack.org/covertype.test.arff.
+mlpack::data::Load("covertype.test.arff", testDataset, true);
+
+// Predict class of first test point.
+const size_t firstPrediction = gb.Classify(testDataset.col(0));
+std::cout << "Predicted class of first test point is " << firstPrediction << "."
+    << std::endl;
+
+```
+
+---
```

Comment:
Do you think we could add additional examples?  The main things I would want to show would be:

 * Inspect a `DecisionTreeRegressor` (maybe print some information about it, or maybe collect the predictions for each weak learner for a point)
 * Serialize the model to disk (and then load it from disk)
 * Predict points in batch
 * Build a `GradientBoosting` model on data of a different type (e.g. `arma::fmat` or `arma::sp_fmat` or whatever).